packages/mux-player-react/src/lazy.tsx

```diff
@@ -1,117 +1,71 @@
-import React, { useEffect, useState } from 'react';
-import type { DetailedHTMLProps, HTMLAttributes } from 'react';
+import React, { useRef } from 'react';
 
 import ConditionalSuspense from './ConditionalSuspense';
 import useIsBrowser from './useIsBrowser';
 import useIsIntersecting from './useIsIntersecting';
 
 import type { MuxPlayerProps, MuxPlayerRefAttributes } from './index';
 
-interface MuxPlayerElement extends DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement> {
-  nohotkeys?: boolean | undefined;
-}
-
-declare global {
-  // eslint-disable-next-line @typescript-eslint/no-namespace
-  namespace JSX {
-    interface IntrinsicElements {
-      'mux-player': MuxPlayerElement;
-    }
-  }
-}
-
 const MuxPlayerIndex = React.lazy(() => import('./index'));
 
-interface FallbackProps extends MuxPlayerProps {
-  onIntersection?: () => void;
+interface MuxPlayerLazyProps extends MuxPlayerProps {
+  loading?: 'page' | 'viewport';
 }
-const Fallback = (props: FallbackProps) => {
-  const { style, className, onIntersection, placeholder } = props;
+const MuxPlayer = React.forwardRef<MuxPlayerRefAttributes, MuxPlayerLazyProps>((props, ref) => {
+  const { loading = 'viewport', ...playerProps } = props;
 
-  const intersectionRef = React.useRef<HTMLElement>(null);
+  // We load mux player once two conditions are met:
+  // 1. We're in a browser (react.lazy doesn't work on the server in react 17)
+  const isBrowser = useIsBrowser();
+  // 2. The player has entered the viewport, according to the fallback (if enabled).
+  const intersectionRef = useRef<HTMLDivElement>(null);
```

Comment:
suggestion(non-blocking): If possible, we should avoid the computations required for `isIntersecting` if `loading === 'page'`